src/routes/docs/tutorials/android/step-4/+page.markdoc

```diff
@@ -0,0 +1,213 @@
+---
+layout: tutorial
+title: Add authentication
+description: Add Appwrite authentication to you Android app.
+step: 4
+---
+
+# Creating an account service {% #creating-an-account-service %}
+
+We can use services to abstract business logic from our views.
+Create a service to handle user authentication with a new file `services/AccountService.kt`.
+
+Add the following code to it.
+
+```kotlin
+package <YOUR_ROOT_PACKAGE_HERE>.services
+
+import io.appwrite.Client
+import io.appwrite.ID
+import io.appwrite.models.User
+import io.appwrite.exceptions.AppwriteException
+import io.appwrite.services.Account
+
+class AccountService(client: Client) {
+    private val account = Account(client)
+
+    suspend fun getLoggedIn(): User<Map<String, Any>>? {
+        return try {
+            account.get()
+        } catch (e: AppwriteException) {
+            null
+        }
+    }
+
+    suspend fun login(email: String, password: String): User<Map<String, Any>>? {
+        return try {
+            account.createEmailSession(email, password)
+            getLoggedIn()
+        } catch (e: AppwriteException) {
+            null
+        }
+    }
+
+    suspend fun register(email: String, password: String): User<Map<String, Any>>? {
+        return try {
+            account.create(ID.unique(), email, password)
+            login(email, password)
+        } catch (e: AppwriteException) {
+            null
+        }
+    }
+
+    suspend fun logout() {
+        account.deleteSession("current")
```

Comment:
```suggestion
        try {
            account.deleteSession("current")
        } catch (ex: AppwriteException) {
            // Handle logout failure
        }
```

Reply:
@abnegate We're gonna avoid this for sake of brevity in this tutorial for now.